README.md

```diff
@@ -1,153 +1,169 @@
+- **Configuration Repository**: A central location for hosting node configuration files.
+The configuration is used to automate the deployment of nodes.
+- **Configuration Push**: The configuration of nodes is pushed from Cluster Manager to nodes.
+- **Configuration Cleanup**: Configuration files, service, packages, etc. are removed from nodes.
+- **Configuration Verification**: Checks to ensure that the node configuration is truly functional.
+- **Configuration Upgrade**: Automates the process of upgrading node software components.
+Upgrades can be automatically triggered and can be performed cluster-wide or rolling.
+- **Role/Group-Based Configuration**: Nodes can be assigned a group or role.
+Role/Group-Based configuration selectively manages services on nodes statically by the
+operator or dynamically based on service availability policy.
+
+**Note:** Dynamic role assignment is **not** supported in initial release of Cluster Manager.
 
 ##Bootstrap
-Bootstraping involves bringing up a node or cluster for the first time.
+Bootstrap involves provisioning a node or cluster for the first time.
 
 ###Node
-The node bootstrap requires:
-- installing the base image
-- performing first time configurations like disk partitioning
-- allocating an IP for node to become reachable to rest of the cluster management.
-- setting up user and credentials with enough permissions to perform configuration management.
-- starting the pre-requisite services like Serf
+Cluster Manager provides the following node bootstrapping capabilities:
+- Installing the base image
+- Performing initial configurations such as disk partitioning
+- Assigning an IP address
+- Configuring user credentials and permissions to perform configuration management tasks
+- Starting infrastructure services such as Serf
  
 **Note:** To start simple this feature is **not** provided as part of initial
-release of cluster manager. In the initial release the user shall be provided the base image
-as a *bootable iso* file that they will need to integrate with their node onborading process.
+release of cluster manager. In the initial release the user will be provided the base image
+as a *bootable iso* file that can be integrated into an existing node onborading process.
 
 ###Cluster
-The cluster bootstrap requires adding the first bootstrapped node to the cluster by starting
-the cluster management service on it with appropriate parameters like:
-- configuration management parameters like user information, configuration repository
-- inventory management parameters like database url
+The cluster bootstrap installs and configures clustering software components such as swarm-master
+(for Docker Swarm) to the first bootstrapped node with parameters such as:
+- Configuration management parameters such as user information, configuration repository
+- Inventory management parameters such as database url
 
 **Note:** To start simple this feature is provided in form of a bootstrap script that the
```

Comment:
OK, I'll remove. As for setting ```ucp_bootstrap_node_name```, lets address that in the install/operations guide.